warehouse/query-core/src/main/java/datawave/query/jexl/visitors/FieldToFieldComparisonVisitor.java

```diff
@@ -0,0 +1,69 @@
+package datawave.query.jexl.visitors;
+
+import datawave.query.jexl.JexlASTHelper;
+import org.apache.commons.jexl2.parser.*;
+
+public class FieldToFieldComparisonVisitor extends RebuildingVisitor {
+    /**
+     * force evaluation for field to field comparison
+     *
+     * @param root
+     * @return
+     */
+    public static ASTJexlScript forceEvaluationOnly(JexlNode root) {
+        FieldToFieldComparisonVisitor vis = new FieldToFieldComparisonVisitor();
+        return (ASTJexlScript) root.jjtAccept(vis, null);
+    }
+    
+    /**
+     * detect identifier on both sides of nodes and wrap it with evaluation-only reference
+     *
+     * @param node
+     * @return
+     */
+    protected JexlNode evaluationOnlyForFieldToFieldComparison(JexlNode node) {
+        int identifierNodes = 0;
+        
+        // check both sides of nodes and count the nodes with identifier(s)
+        for (int i = 0; i < node.jjtGetNumChildren(); i++) {
+            if (JexlASTHelper.getIdentifiers(node.jjtGetChild(i)).size() > 0) {
```

Comment:
Is it worth checking to see if this node has two children exactly? Based on the entry points below I don't think we would run into the case where num_children > 2.

Can you add a test for "FIELD_A == FIELD_B == FIELD_C"? If only to document that we do not support this case.

Reply:
I updated the unit test which just throws Parser exception. Thanks!